helm/revoke.go

```diff
@@ -85,54 +114,101 @@ func revokeAccessFromRBACEntities(
 ) error {
 	logger := logging.GetProjectLogger()
 
+	errList := NewRevocationErrors()
 	numEntities := len(rbacEntities)
 	for idx, rbacEntity := range rbacEntities {
 		logger.Infof("Revoking access from entity %s (%d of %d entities)", rbacEntity, idx+1, numEntities)
+		roleName := getTillerAccessRoleName(rbacEntity.EntityID(), tillerNamespace)
+
+		logger.Infof("Deleting entity role %s", roleName)
+		if err := deleteEntityRole(kubectlOptions, tillerNamespace, roleName, rbacEntity); err != nil {
+			errList.AddError(err)
+			logger.Warningf("Unable to delete role for entity %s from namespace %s", rbacEntity.Subject().Name, tillerNamespace)
+		} else {
+			logger.Infof("Deleted role for entity %s", rbacEntity.Subject().Name)
+		}
 
-		logger.Infof("Deleting entity role and binding")
-		if err := deleteEntityRoleAndBinding(kubectlOptions, tillerNamespace, rbacEntity); err != nil {
-			logger.Errorf("Unable to delete role for entity %s from namespace %s", rbacEntity.Subject().Name, tillerNamespace)
-			return err
+		logger.Infof("Deleting entity role binding")
+		if err := deleteEntityRoleBinding(kubectlOptions, tillerNamespace, roleName, rbacEntity); err != nil {
+			errList.AddError(err)
```

Comment:
Can you check if the error returned is the "Not exists" error, and then only treat it as an error if it is an error that does not match that one?

For example, if you use a specific error type, you can do the check as:

```go
err := deleteEntityRoleBinding(kubectlOptions, tillerNamespace, roleName, rbacEntity)
if err == nil {
    logger.Infof("Deleted role binding for entity %s", rbacEntity.Subject().Name)
} else if err, ok := err.(*RoleBindingDoesNotExist); ok {
    logger.Warningf("RoleBinding does not exist. Ignoring error and continuing.")
} else {
    return err
}
```

Note that I didn't do the errlist here. I actually originally meant the error aggregation should happen at the top loop. E.g (in pseudo code):

```
err = revokeAccessFromRBACEntity(rbacEntity)
if err {
    errList.AddError(err)
}
```

This way, we attempt to revoke every entity in the list, but short circuit attempts within each entity (so some entities will be potentially partially revoked). The goal in this suggestion is that we want to try to revoke all the entities the user requested to do, so that they only retry the ones that failed.

Reply:
Makes good sense. I will implement this. Thanks!